### DISCUSSION TITLE
Removing groovy lint

### DISCUSSION THREAD

**Discussion by marchoeppner** [bio-raum/FooDMe2#58] [General]
Hi,

as mentioned, the groovy linter seems to be acting up. More importantly, current versions start tripping over some nextflow'isms that cannot be properly "caught" with exceptions - most importantly something like

```NEXTFLOW
include { MULTIQC as MULTIQC_TEST } form './../../modules/multiqc'
```

I am leaning towards removing the groovy linter from the github workflows to prevent failiures. It was always a bit of a crummy workaround since Nextflow isn't exactly Groovy. But it used to work pretty well...not anymore ;)

Would that be ok with you?

**Comment by gregdenay** [comment]
Do you think we could use this instead? Or are we not 'core' enough?
https://github.com/nf-core/gh-actions-lint?tab=readme-ov-file

**Reply by marchoeppner** [reply]
We do deviate a fair bit from the nf-core template (mostly because a lot of it is meant to increase compatibility with nf-tower and the whole "commercialization" of nf-core. Pretty sure this would fail on our code.

**Reply by gregdenay** [reply]
The i see no other solution than removing the linter.We should check back in few weeks to see if the issues are lifted though

**Reply by marchoeppner** [reply]
So this was fixable after all. I version-locked the linter to v14 as v15 apparently introduced some new functionality that won't behave with Nextflow.

**Reply by gregdenay** [reply]
Great!
